src/classes/HouseholdNamingService.cls

```diff
@@ -98,9 +131,24 @@ public without sharing class HouseholdNamingService {
 
     private void setNumberOfHouseholdMembers(List<SObject> householdsOrAccounts, Map<Id, List<Contact>> membersByHouseholdId) {
         for (SObject household : householdsOrAccounts) {
-            List<Contact> householdMembers = membersByHouseholdId.get(idFor(household));
-            household.put(UTIL_Namespace.StrTokenNSPrefix('Number_of_Household_Members__c'),
-                    householdMembers != null ? householdMembers.size() : 0);
+            List<Contact> members = membersByHouseholdId.get(idFor(household));
+            if (numberOfHouseholdMembersIsIncorrect(household, members)) {
+                refreshNumberOfHouseholdMembers(household, members);
+                unitOfWorkInst.registerDirty(new List<SObject>{household});
+            }
+        }
+    }
+
+    private void refreshNumberOfHouseholdMembers(SObject household, List<Contact> members) {
+        household.put(UTIL_Namespace.StrTokenNSPrefix('Number_of_Household_Members__c'),
```

Comment:
(Non Blocking!) Not sure I love this use of inline string for field name, but I can see how `SObjectType.<SObjectName>.fields.Number_Of_Househodl_Members__c.Name` is difficult here

Reply:
Yeah, I don't love this either, but hesitant to change it at the moment.  I think we need a follow up WI to define the interface IHouseholds that NPSP_Households and NPSP_HouseholdAccounts will implement.  Then we can define behavior like this on the interface and have each implementation contain its own "numberOfHouseholdMembers(Integer count)" method implementation, then removing this usage of UTIL_Namespace and the string field reference.